Dashboard report widget

kaapstorm opened the pull request:
## Product Description

This change allows users to show UCRs in the Campaign Dashboard.

![image](https://github.com/user-attachments/assets/de4839fa-0b41-4215-8492-cc73cac691f7)

If that "Report Filters" form looks a little funky, that would be because it is. It is using half-Bootstrap5-half-Bootstrap3 elements. The input fields don't work. The buttons don't work. More on that below.

## Technical Summary

Jira: [SC-4291](https://dimagi.atlassian.net/browse/SC-4291)

#### :scream: So many commits?! 

The `nh/dash/report_widget` builds on top the following PRs:

* https://github.com/dimagi/commcare-hq/pull/36043
* https://github.com/dimagi/commcare-hq/pull/36048
* https://github.com/dimagi/commcare-hq/pull/36089

These are the commits you are looking for:

1. 292c9f3a6d4 Move report content into partial template
2. 66ba6952d3f Add properties to model that hqReport needs
3. b3b59e39db8 Add report data to view context
4. 13cc5e55753 Add template for dashboard reports
5. 6201f3f5493 Front-end support for report widget
6. ed5a9a99b86 Feeble attempts to get filters to work with BS5

@biyeun @orangejenny please could you take a look at that last commit, and tell me, can I get UCRs to work with Bootstrap 5 ... in the next 8 hours ... (and if so, how) or should I backport the Campaign Dashboard feature to Bootstrap 3 ... (also in the next 8 hours)? (We have a deadline to meet.)

## Feature Flag

campaign_dashboard

## Safety Assurance

### Safety story

Tested locally.

### Automated test coverage

Models are covered by tests. Coverage of views is partial. UI elements are not covered by tests.

### QA Plan

QA plan pending.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change


[SC-4291]: https://dimagi.atlassian.net/browse/SC-4291?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

kaapstorm committed:
Set "html_id_suffix" in report view context

kaapstorm committed:
Add "html_id_suffix" to HTML IDs in BS5 JavaScript

kaapstorm committed:
Add "html_id_suffix" to HTML IDs in BS3 JavaScript

kaapstorm committed:
"Bootstrap 5 Migration - Rebuilt diffs"

kaapstorm committed:
Add 'html_id_suffix' to initial page data

kaapstorm committed:
Add "html_id_suffix" to HTML IDs in BS3 templates

kaapstorm committed:
Add "html_id_suffix" to HTML IDs in BS5 templates

kaapstorm committed:
Use the methods set on `self`, not their defaults.

Prior to this change, self.getReportBaseUrl() and self.getReportParams() are
unused. This change treats them the same as getReportRenderUrl(), as you
would expect from the code.

kaapstorm committed:
Use report URL for UCRs

kaapstorm committed:
"Bootstrap 5 Migration - Rebuilt diffs"

kaapstorm committed:
Move report content into partial template

kaapstorm committed:
Add "html_id_suffix" to `js_options`

kaapstorm committed:
Revert "Add 'html_id_suffix' to initial page data"

This reverts commit 64bddb2eedf448e0dec385307dac4ee88d579d2a.

kaapstorm committed:
Use html_id_suffix in js_options

kaapstorm committed:
Add to initial page data in base templates

kaapstorm committed:
"Bootstrap 5 Migration - Rebuilt diffs"

kaapstorm committed:
Support html_id_suffix in maps_utils.js

kaapstorm committed:
"Bootstrap 5 Migration - Rebuilt diffs"

kaapstorm committed:
Add properties to model that hqReport needs

kaapstorm committed:
Add report data to view context

kaapstorm committed:
Add template for dashboard reports

kaapstorm committed:
Front-end support for report widget

mkangia commented:
Hey @kaapstorm 
Just confirming that the 6 commits you linked in the description are the relevant ones to review? rest all are from the other PRs.

I assume this PR will get merged after other 3 are merged?

kaapstorm commented:
> Just confirming that the 6 commits you linked in the description are the relevant ones to review? rest all are from the other PRs.

Yes, that's right.

> I assume this PR will get merged after other 3 are merged?

:+1: Yeah.

I see there are already 4 conflicts. I'm going to be more and more tempted to rebase, but I'm still keen to hear your thoughts on those 6 commits, so I can follow up feedback asap.

kaapstorm committed:
Missing suffix

Fixes Filter Options accordion in UCRs

kaapstorm committed:
"Bootstrap 5 Migration - Rebuilt diffs"

kaapstorm commented:
Regarding new commits:
* [Missing suffix](https://github.com/dimagi/commcare-hq/pull/36093/commits/887eda842d4fb0a7f7de9443b1f404a30b3f24ee) has been added to PR https://github.com/dimagi/commcare-hq/pull/36043 where it belongs.
* [Revert to Bootstrap 3](https://github.com/dimagi/commcare-hq/pull/36093/commits/b43d86a85e702151e0696c59a41e800664e93d99) is PRed separately here: https://github.com/dimagi/commcare-hq/pull/36097
* [fixup! Front-end support for report widget](https://github.com/dimagi/commcare-hq/pull/36093/commits/9c0c293ba86d080c9a7c5871aee41a14bc2257e7) would love to be rebased.

kaapstorm committed:
Just get Clause 3.7 Sonnet to do it all again

Here's a summary of what we changed:

1. Updated the base template from `hqwebapp/bootstrap5/base_section.html` to `hqwebapp/bootstrap3/base_section.html`.

2. Replaced Bootstrap 5 utility classes with Bootstrap 3 equivalents:
   - `d-flex`, `justify-content-between`, `align-items-center`, `mb-3` → `clearfix margin-bottom-10`, `pull-left`, `pull-right`
   - `d-none` → `hide`
   - `me-1` → `margin-right-5`
   - `mb-4` → `margin-bottom-20`
   - `p-3` → `padding-10`
   - `p-0` → `no-padding`
   - `g-0` → removed (no direct equivalent)
   - `border-end` → `border-right`

3. Updated nav/tab components:
   - Removed `.nav-item` and `.nav-link` classes
   - Changed `data-bs-toggle="tab"` to `data-toggle="tab"`

4. Updated modal components:
   - Changed `data-bs-dismiss="modal"` to `data-dismiss="modal"`
   - Changed `data-bs-toggle="modal"` to `data-toggle="modal"`
   - Changed `data-bs-target="#widget-modal"` to `data-target="#widget-modal"`
   - Replaced `btn-close` with `close` and added `<span aria-hidden="true">&times;</span>`
   - Moved modal title from `<h5>` to `<h4>` and repositioned it after the close button

5. Updated cards to panels:
   - `card` → `panel panel-default`
   - `card-header` → `panel-heading`
   - `card-body` → `panel-body`

6. Updated form components:
   - `form-check` → `checkbox`
   - Removed `form-check-label` class

7. Updated JavaScript references:
   - Changed import from 'reports/js/bootstrap5/base' to 'reports/js/bootstrap3/base'
   - Updated event handlers from `shown.bs.tab` to use Bootstrap 3's events
   - Updated selectors like `.nav-tabs .nav-link.active` to `.nav-tabs .active a`
   - Updated DOM manipulation methods for showing/hiding elements from `.classList.add('d-none')` to Bootstrap 3's `.addClass('hide')`

8. Removed all `use_bootstrap5` decorators from view functions

9. Updated include references from Bootstrap 5 to Bootstrap 3:
   - Changed `hqwebapp/htmx/error_modal.html` to `hqwebapp/bootstrap3/error_modal.html`
   - Changed `reports/standard/partials/bootstrap5/filter_panel.html` to `reports/standard/partials/bootstrap3/filter_panel.html`

kaapstorm committed:
`js_entry` to `js_entry_b3 "campaign/js/dashboard"`

I am not sure how testing succeeded without this?! Caching? Something
slow to restart?

kaapstorm commented:
YOLO. Rebased.

![image](https://github.com/user-attachments/assets/d172544f-9e58-456d-890a-0729e48f64a8)

Still the same issue with filter input fields. @biyeun please :sob: what am I missing?

Bootstrap 3 PR is here: https://github.com/dimagi/commcare-hq/pull/36100

Once again, relevant commits:

* 739da7d6fcd Move report content into partial template
* 04d2d01eddb Add properties to model that hqReport needs
* 3a7b75b05ca Add report data to view context
* 75a742015be Add template for dashboard reports
* f94b7bed4fc Front-end support for report widget

kaapstorm commented:
The backport to BS3 is not necessary. I am going to close this PR, follow up on feedback, and open a new PR.
